test/unit/vl-radio.test.html

```diff
@@ -0,0 +1,161 @@
+        should.not.exist(radioHiddenLabel);
+        radio.setAttribute('single', true);
+        radioHiddenLabel = radio.shadowRoot.querySelector('.vl-u-visually-hidden');
+        should.exist(radioHiddenLabel);
+      });
+
+      test('het klikken op het input veld zal de radio checked waarde niet meer aanpassen bij een radio met een boolean checked attribuut', () => {
+        const radio = fixture('vl-radio-fixture');
+        const radioInput = radio.shadowRoot.querySelector('input');
+        assert.isFalse(radioInput.checked);
+        radioInput.click();
+        assert.isTrue(radioInput.checked);
+        radioInput.click();
+        assert.isTrue(radioInput.checked);
+      });
+
+      test('toggle zal de checked waarde aanpassen bij een radio met een boolean checked attribuut, behalve als die al checked was', () => {
+        const radio = fixture('vl-radio-fixture');
+        const radioInput = radio.shadowRoot.querySelector('input');
+        assert.isFalse(radioInput.checked);
+        radio.toggle();
+        assert.isTrue(radioInput.checked);
+        radio.toggle();
+        assert.isTrue(radioInput.checked);
+      });
+
+      test('het klikken op het input veld zal een event gooien met de waarde van het boolean checked attribuut', (done) => {
+        const radio = fixture('vl-radio-fixture');
+        radio.addEventListener('input', (e) => {
+          assert.isTrue(e.srcElement.checked);
+          done();
+        });
+        radio.shadowRoot.querySelector('input').click();
+      });
+
+      test('toggle zal een event gooien met de waarde van het boolean checked attribuut', (done) => {
+        const radio = fixture('vl-radio-fixture');
+        radio.addEventListener('input', (e) => {
+          assert.isTrue(e.srcElement.checked);
+          done();
+        });
+        radio.toggle();
+      });
+
+      test('het klikken op het input veld zal slechts één input event doorgestuurd worden', (done) => {
+        const radios = fixture('vl-radios-fixture');
+        const radio = radios[0];
+        const radioInput = radio.shadowRoot.querySelector('input');
+        let counter = 0;
+        radio.addEventListener('input', (e) => {
+          counter++;
+          if (e.srcElement.checked) {
+            setTimeout(() => {
+              assert.equal(counter, 1);
+              done();
+            });
+          }
+        });
+        radioInput.click();
+      });
+
+      test('het klikken op het input veld zal het origineel change event doorsturen', (done) => {
+        const radios = fixture('vl-radios-fixture');
+        const radio = radios[0];
+        const radioInput = radio.shadowRoot.querySelector('input');
+        radioInput.addEventListener('change', (e) => {
+          done();
+        });
+        radioInput.click();
+      });
+
+      test('het klikken op een radio en daarna een andere zal de originele terug afzetten', () => {
```

Comment:
er zijn 3 radio's, moeten we niet testen dat er steeds maar 1 checked is?

Reply:
Best wel, ik pas het aan.